### PR TITLE
Endpoints to grant/revoke admin access

### PR DESCRIPTION
**Pull request by kevinemoore**
## Description
Adding admin-only endpoints to the registry to grant & revoke admin access for other users.

- [ x ] Unit Tests

### CODE REVIEW

File: registry/tests/auth_test.py [hunk 1]
```diff
@@ -341,6 +341,45 @@ def testCompilerLogin(self):
         assert 'expires_at' in refreshed_token_payload
         assert refreshed_token_payload['access_token'] == refreshed_token_payload['refresh_token']
 
+    def testGrantRevokeAdmin(self):
+
+        def list_users(token):
+            response = self.app.get(
```
Comment (meffij) [inline registry/tests/auth_test.py]: just return the result of the call (i.e. `return self.app.get(...`)

File: registry/quilt_server/views.py [hunk 2]
```diff
@@ -2206,6 +2206,37 @@ def enable_user():
...
         raise ApiException(requests.codes.not_found, ex.message)
 
+@app.route('/api/users/grant_admin', methods=['POST'])
+@api(enabled=ENABLE_USER_ENDPOINTS, require_admin=True, schema=USERNAME_SCHEMA)
+@as_json
+def grant_admin():
+    try:
+        data = request.get_json()
+        username = data['username']
+        user = User.query.filter_by(name=username).with_for_update().one_or_none()
```
[the remark this answers is not in the document]
Comment (kevinemoore) [reply registry/quilt_server/views.py]: I wrote the code that way to match `_enable_user` and `_disable_user`. Would you like me to change those too? I think we should keep them consistent.

File: registry/quilt_server/views.py [hunk 3]
```diff
@@ -2206,6 +2206,37 @@ def enable_user():
...
+        _grant_admin(user)
+        db.session.commit()
+        return {}
+    except NotFoundException as ex:
+        raise ApiException(requests.codes.not_found, ex.message)
```
[the remark this answers is not in the document]
Comment (kevinemoore) [reply registry/quilt_server/views.py]: Same comment as above about consistency with enable/disable user.

File: registry/quilt_server/views.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (meffij) [reply registry/quilt_server/views.py]: +1 for consistency

**Pull request merged**